src/compiler/parser.ts

```diff
@@ -7168,8 +7169,19 @@ namespace Parser {
                 //
                 // could be legal, it would add complexity for very little gain.
                 case SyntaxKind.InterfaceKeyword:
-                case SyntaxKind.TypeKeyword:
                     return nextTokenIsIdentifierOnSameLine();
+                case SyntaxKind.TypeKeyword:
+                    const isIdentifierOnSameLine = nextTokenIsIdentifierOnSameLine();
+                    if (isIdentifierOnSameLine && token() === SyntaxKind.AsKeyword || token() === SyntaxKind.SatisfiesKeyword) {
+                        return lookAhead(() => {
+                            nextToken();
+                            parseTypeParameters();

```

Comment:
This feels like it would be pretty wasteful, but I suppose that people can't write this code today anyway and it'd only be paid for `as` and `satisfies`.